Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/transport/round_trippers.go

```diff
@@ -130,6 +133,36 @@ func (rt *basicAuthRoundTripper) CancelRequest(req *http.Request) {
 
 func (rt *basicAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
 
+type actAsRoundTripper struct {
+	actAs string
+	rt    http.RoundTripper
+}
+
+// NewActAsRoundTripper will apply a BASIC auth authorization header to a
+// request unless it has already been set.
+func NewActAsRoundTripper(actAs string, rt http.RoundTripper) http.RoundTripper {
+	return &actAsRoundTripper{actAs, rt}
+}
+
+func (rt *actAsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(req.Header.Get("Act-As")) != 0 {
+		return rt.rt.RoundTrip(req)
+	}
+	req = cloneRequest(req)
+	req.Header.Set("Acting-As", rt.actAs)
```

Comment:
Other REST APIs I could find that had impersonation:

- something called [redmine](http://www.redmine.org/projects/redmine/wiki/Rest_api#User-Impersonation) uses `X-Redmine-Switch-User: jsmith`, only admin user can do it.
- something called[telligent](https://community.telligent.com/community/f/533/t/1065840) uses `Rest-Impersonate-User: jsmith`
- [HBase](https://issues.apache.org/jira/browse/HBASE-13115) uses `doAs: jsmith`